app/models/registration_payment.rb

```diff
@@ -29,7 +29,11 @@ def payment_status
     end
   end
 
-  private def attempt_auto_close
-    registration.competition.attempt_auto_close!
+  private def auto_close_hook
+    if amount_lowest_denomination > 0
+      return registration.consider_auto_close
+    else
+      return false
+    end
```

Comment:
FWIW you don't need the `if` here because I already added an `unless:` check in the `after_create` routine

Reply:
oh nice!